### ISSUE TITLE
IOError: could not get source code

### ISSUE THREAD

**Issue by denfromufa** [ipython/traitlets#189]
packaging ipython with pyinstaller:

```
Traceback (most recent call last):
  File "<string>", line 208, in <module>
  File "site-packages\IPython\core\display.py", line 152, in display
  File "site-packages\traitlets\config\configurable.py", line 404, in instance
  File "site-packages\traitlets\traitlets.py", line 910, in __new__
  File "site-packages\traitlets\traitlets.py", line 934, in setup_instance
  File "site-packages\traitlets\traitlets.py", line 925, in setup_instance
  File "site-packages\traitlets\traitlets.py", line 1563, in instance_init
  File "site-packages\traitlets\traitlets.py", line 492, in instance_init
  File "site-packages\traitlets\traitlets.py", line 483, in _dynamic_default_cal
lable
  File "site-packages\traitlets\traitlets.py", line 109, in _deprecated_method
  File "inspect.py", line 690, in getsourcelines
  File "inspect.py", line 538, in findsource
IOError: could not get source code



alabaster (0.7.6)
argh (0.26.1)
ashiba (0.0)
astroid (1.3.8)
Babel (2.1.1)
backports-abc (0.4)
backports.ssl-match-hostname (3.4.0.
beautifulsoup4 (4.4.1)
binstar (0.11.0)
blessings (1.6)
bokeh (0.10.0)
bpython (0.14.2)
bqplot (0.4.2)
certifi (2015.9.6.2)
chest (0.2.3)
cloud (2.8.5)
cloudpickle (0.1.1)
clyent (0.3.4)
colorama (0.3.3)
comtypes (1.1.2)
conda (3.19.3)
conda-build (1.18.0)
conda-env (2.4.5)
CoolProp (5.1.2)
curses (2.2)
curtsies (0.1.21)
cycler (0.10.0)
Cython (0.23.4)
dask (0.8.0)
decorator (4.0.6)
DerApproximator (0.52)
dill (0.2.4)
docopt (0.6.2)
docutils (0.12)
dolo (0.5.dev0)
enum (0.4.4)
enum34 (1.1.2)
et-xmlfile (1.0.1)
Flask (0.10.1)
FuncDesigner (0.5404)
funcsigs (0.4)
functools32 (3.2.3.post2)
futures (3.0.3)
gevent (1.0.1)
gevent-websocket (0.9.3)
greenlet (0.4.9)
HeapDict (1.0.0)
html5lib (0.999)
ipdb (0.8.1)
ipycache (0.1.4)
ipykernel (4.2.2)
ipython (4.0.1)
ipython-genutils (0.1.0)
ipywidgets (4.1.1)
itsdangerous (0.24)
jdcal (1.0)
jedi (0.9.0)
Jinja2 (2.8)
JSAnimation (0.1)
jsonpickle (0.9.2)
jsonschema (2.5.1)
jupyter (1.0.0)
jupyter-client (4.1.1)
jupyter-console (4.0.3)
jupyter-core (4.0.6)
jupyter-kernel-gateway (0.4.0.dev0)
llvmlite (0.8.0)
locket (0.2.0)
logilab-common (1.1.0)
lxml (3.4.4)
MarkupSafe (0.23)
matplotlib (1.5.1)
mccabe (0.3.1)
menuinst (1.3.2)
mistune (0.7.1)
mpmath (0.19)
nbconvert (4.1.0)
nbformat (4.0.1)
networkx (1.10)
nose (1.3.7)
notebook (4.1.0)
numba (0.23.1)
numexpr (2.4.4)
numpy (1.10.4)
numpydoc (0.5)
oauthlib (1.0.3)
oletools (0.44)
openopt (0.5404)
openpyxl (2.3.0)
pandas (0.17.0)
partd (0.3.2)
path.py (8.1.2)
pathtools (0.1.2)
patsy (0.4.0)
pickleshare (0.6)
Pillow (3.0.0)
Pint (0.6)
pip (8.0.3)
pipwin (0.2.4)
plumbum (1.6.1.post0)
ply (3.8)
poster (0.8.1)
prompt-toolkit (0.57)
pss (1.40)
psutil (3.4.2)
ptpython (0.28)
ptvsd (2.2.0)
py-trello (0.4.3)
pycosat (0.6.1)
Pygments (2.0.2)
pygraphviz (1.3rc2)
PyInstaller (3.1)
pylint (1.4.4)
Pympler (0.4.2)
pymssql (2.1.1)
pyodbc (3.0.10)
Pyomo (4.2.10784)
pyomo.extras (2.0)
pyparsing (2.0.3)
PyPDF2 (1.24)
PyPrind (2.9.3)
pypyodbc (1.3.3)
PyQt4 (4.11.4)
pyreadline (2.1)
Pyro (3.16)
PySide (1.2.4)
Python-contrib-nbextensions (alpha)
python-dateutil (2.4.2)
pythonnet (2.1.0.dev1)
pythreejs (0.1.17)
pytz (2015.7)
PyUtilib (5.2.3601)
pywin32 (220)
PyYAML (3.11)
pyzmq (14.7.0)
qgrid (0.2.0)
qtconsole (4.1.0)
reportlab (3.1.44)
requests (2.9.1)
requests-oauthlib (0.6.0)
robobrowser (0.5.3)
rodeo (0.4.4)
scandir (1.1)
scikit-learn (0.16.1)
scipy (0.16.1)
seaborn (0.7.0)
setuptools (19.2)
simplegeneric (0.8.1)
singledispatch (3.4.0.3)
six (1.10.0)
snowballstemmer (1.2.0)
Sphinx (1.3.1)
sphinx-rtd-theme (0.1.8)
spyder (2.3.7)
SQLAlchemy (1.0.9)
statsmodels (0.6.1)
suds-jurko (0.6)
sympy (0.7.6.1)
tabulate (0.7.5)
toolz (0.7.4)
tornado (4.3)
traitlets (4.0.0)
ujson (1.33)
uncertainties (2.4.7.1)
watchdog (0.8.3)
wcwidth (0.1.5)
Werkzeug (0.10.4)
wheel (0.29.0)
xlrd (0.9.4)
xray (0.6.1)

```

**Comment by denfromufa** [comment]
this looks fixed in traitlets (4.1.0)

**Comment by yishenggudou** [comment]
```
Failed to execute script cli
Traceback (most recent call last):
  File "cli.py", line 591, in <module>
  File "cli.py", line 219, in __init__
  File "cli.py", line 297, in buildProduct
  File "cli.py", line 316, in buildThirdPartyJs
  File "cli.py", line 527, in compress
  File "site-packages/slimit/minifier.py", line 37, in minify
  File "site-packages/slimit/parser.py", line 55, in __init__
  File "site-packages/slimit/lexer.py", line 98, in __init__
  File "site-packages/slimit/lexer.py", line 102, in build
  File "site-packages/ply/lex.py", line 906, in lex
  File "site-packages/ply/lex.py", line 580, in validate_all
  File "site-packages/ply/lex.py", line 822, in validate_rules
  File "site-packages/ply/lex.py", line 833, in validate_module
  File "inspect.py", line 690, in getsourcelines
  File "inspect.py", line 538, in findsource
IOError: could not get source code
```
